Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListImplementation/Rendering/BocListColumnIndexProviderTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using Moq;
+using NUnit.Framework;
+using Remotion.ObjectBinding.Web.UI.Controls;
+using Remotion.ObjectBinding.Web.UI.Controls.BocListImplementation.Rendering;
+
+namespace Remotion.ObjectBinding.Web.UnitTests.UI.Controls.BocListImplementation.Rendering
+{
+  [TestFixture]
+  public class BocListColumnIndexProviderTest
+  {
+    [Test]
+    public void Create_DoesNotCreateLookup ()
+    {
+      // Explicit strict mock to ensure it is not used in the constructor
+      var columnRendererMock = new Mock<IReadOnlyList<BocColumnRenderer>>(MockBehavior.Strict);
+
+      Assert.That(() => new BocListColumnIndexProvider(columnRendererMock.Object), Throws.Nothing);
+    }
+
+    [Test]
+    public void GetColumnIndex ()
+    {
+      var stubColumnDefinition1 = new StubColumnDefinition();
+      var bocColumnRenderer1 = CreateBocColumnRenderer(stubColumnDefinition1, 1, -1);
+
+      var stubColumnDefinition2 = new StubColumnDefinition();
+      var bocColumnRenderer2 = CreateBocColumnRenderer(stubColumnDefinition2, 2, -2);
+
+      var columnIndexProvider = new BocListColumnIndexProvider(new[] { bocColumnRenderer1, bocColumnRenderer2 });
+
+      Assert.That(columnIndexProvider.GetColumnIndex(stubColumnDefinition2), Is.EqualTo(2));
```

Comment:
![image](https://github.com/re-motion/Framework/assets/5082673/1f4e7003-2074-4525-851a-e2b3dcb6bfd1)

It is plenty obvious with the parameter hints. We still need to talk about them but I think that is scheduled for tomorrow anyways.

The off-by-one problem is not really a problem here since we do not do any column index operation. We find the `BocColumnDefinition` instance and return the `ColumnIndex`/`VisibleColumnIndex`. An off by one error is not really applicable so I would keep the current setup. IMO it is clear what is happening without any changes.